agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,283 @@
+    fi
+done
+
+while :; do
+    read -p "Specify Redis port (or press enter for default port '${REDIS_PORT_DEFAULT}'): "
+    let REDIS_PORT=${REPLY:-${REDIS_PORT_DEFAULT}}
+    if [[ $REDIS_PORT -ne 0 ]]
+    then
+        break
+    else
+        printf -- "Port must be a non-zero numeric value; please retry\n"
+    fi
+done
+
+export REDIS_HOST REDIS_PORT
+export PBENCH_REDIS_SERVER="${REDIS_HOST}:${REDIS_PORT}"
+
+function wait_keypress {
+    echo "Press any key to continue"
+    while [[ true ]]; do
+        read -t ${1} -n 1
+        if [[ ${?} = 0 ]]; then
+            return 0
+        else
+            echo "waiting for the keypress"
+        fi
+    done
+}
+
+clear
+
+printf -- "We will now begin the guided pbench image test process. \n\n\n"
+
+wait_keypress 120
+
+# If somebody wants to write their own pbench-tool-meister-start, great, then
+# that person can change the assumption on where the persistent volume is for
+# the Tool Data Sink, and how the "sysinfo", "init", and "end" commands
+# operate with respect to data storage.
+
+export pbench_run=/var/tmp/test-run-dir
+rm -rf ${pbench_run}
+mkdir ${pbench_run}
+printf -- "${pbench_run}\n" > ${pbench_run}/.path
+export pbench_log=${pbench_run}/pbench.log
+
+printf -- "\n\nThe benchmark run directory will be found inside:\n\n"
+
+find ${pbench_run} -ls
+
+wait_keypress 120
+
+
+# Register tools
+
+printf -- "\nNow, let's pick hosts to collect data from, as well as which tools we wish to register for that host.\n\n"
+
+wait_keypress 120
+
+while :; do
+    read -p "Hostname (or press 'ENTER' for default '$CONTROLLER'): "
+    hostvar=${REPLY:-${CONTROLLER}}
+    if ping -c 1 -W 1 ${hostvar} 2>&1 1>/dev/null
+    then
+        echo $hostvar >> ${pbench_run}/remotes.lis
+        read -p "Another host? (y/n)" anotherhost
+        if [ $anotherhost != "y" ]
+        then
+            break
+        fi
+    else
+        printf -- "Host specified either does not exist or is unreachable, please try again:\n"
+    fi
+done
+printf -- "\nDone selecting hosts:\n"
+cat ${pbench_run}/remotes.lis
+printf -- "\nNow to select tools.\n\n"
+
+# Register the specified tools, will be recorded in
+# ${pbench_run}/tools-v1-default
+while :; do
+    read -p "Tool name (type 'help' to see options): " tool
+    if [ -z "$tool" ]
+    then
+        printf -- "Tool name empty. Please write a tool name.\n"
+    else
+        if [ $tool == "help" ]
+        then
+            printf -- "\nAvailable tools:\n"
+            python3 -c "import sys, json; meta = json.load(open(sys.argv[1])); print('  Transient:', *[f'\t{tool}' for tool in meta['transient'].keys()], '  Persistent:', *[f'\t{tool}' for tool in meta['persistent'].keys()], sep='\n')" /opt/pbench-agent/tool-scripts/meta.json
+        else
+            cmd="pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis"
+            printf -- "\n$cmd\n"
+            $cmd
```

Comment:
Below, you use `set -x` to produce this effect...it would be good to use a consistent idiom.